src/OSPSuite.Core/Domain/Services/ParameterIdentifications/ResidualCalculator.cs

```diff
@@ -34,14 +33,23 @@ public ResidualsResult Calculate(IReadOnlyList<SimulationRunResults> simulations
       {
          if (simulationsResults.Any(x => !x.Success))
             return residualResultWithErrorFrom(simulationsResults);
-        
+
          var simulationColumnsCache = new Cache<string, DataColumn>(x => x.PathAsString, x => null);
          simulationsResults.Select(x => x.Results).Each(repo => simulationColumnsCache.AddRange(repo.AllButBaseGrid()));
 
+         return calculateResidualsResult(allOutputMappings, simulationColumnsCache);
+      }
+
+      private ResidualsResult calculateResidualsResult(IReadOnlyList<OutputMapping> allOutputMappings,
+         Cache<string, DataColumn> simulationColumnsCache)
+      {
          var residualResult = new ResidualsResult();
 
          foreach (var outputMapping in allOutputMappings)
          {
+            if (outputMapping.Output == null)
```

Comment:
How can this be null?
Also, please use allOutputMappings.where(x=>x.Output!=null) in the foreach

Reply:
Simulation can have null Outputs, meaning that the corresponding Observed Data is not mapped to an Output. This is not a valid OutputMapping and should not be taken into account for the calculation

Corrected the foreach and also added the above explanation as a comment

Reply:
I don't understand. Ho w can a simulation mapping be valid if there is no output? IMO this is wrong. Please explain

Reply:
It is not valid, that's why we are skipping it. In the Output Mapping presenter for the simulation we have all the observed data listed and an output can be mapped or not. The outputs that are not mapped ( == null  ) are not being used - here is where they are actually filtered out. Alternatively it could also be done at some earlier point in code.

Reply:
(though not sure which point that could be)

Reply:
>In the Output Mapping presenter for the simulation we have all the observed data listed and an output can be mapped or not

This is a presentation thing. But why are objects created for invalid mapping? Do you want to have a call?

Reply:
@msevestre I see your point. Give me a few minutes to go over the code